assets/js/tools/fonts.js

```diff
@@ -0,0 +1,49 @@
+import FontFaceObserver from 'fontfaceobserver';
+
+export default class FontLoader {
+    fonts = [];
+
+    constructor(fontsArray) {
+        this.fonts = fontsArray;
+    }
+
+    load() {
+        const fontsToLoad = [];
+
+        this.fonts.map((font) => {
+            const fontObserver = font.weight !== undefined
+                ? new FontFaceObserver(font.name, {weight: font.weight}).load(null, 400)
+                : new FontFaceObserver(font.name).load(null, 400);
+
+            fontsToLoad.push(fontObserver);
+        });
+
+        Promise.all(fontsToLoad)
+            .then(() => {
+                // eslint-disable-next-line no-console
+                console.log('All fonts have loaded via FontFaceObserver');
+                this.saveSession();
+            })
+            .catch(() => {
+                this.removeSession();
+            });
+
+        this.setLoadingClass();
+    }
+
+    saveSession() {
+        sessionStorage.fontsLoaded = true;
+
+        this.setLoadingClass();
+    }
+
+    removeSession() {
+        sessionStorage.fontsLoaded = false;
+    }
+
+    setLoadingClass() {
+        if (sessionStorage.fontsLoaded) {
+            document.documentElement.classList.add('fonts-loaded');
+        }
+    }
+}
```

Comment:
Can you add a newline at the end of this file (keeps git happy) - Looks like you can enable this across all files in PHPStorm like so https://stackoverflow.com/a/36020003/2166894